packages/components/src/popover/index.js

```diff
@@ -361,38 +219,109 @@ const Popover = ( {
-	const anchor = useAnchor(
-		anchorRefFallback,
-		contentRef,
-		anchorRect,
-		getAnchorRect,
-		anchorRef,
-		shouldAnchorIncludePadding,
-		anchorVerticalBuffer,
-		anchorHorizontalBuffer
-	);
+	useEffect( () => {
+		const resize = () => setIsMobileViewport( window.innerWidth < 782 );
+		window.addEventListener( 'resize', resize );
+		return () => window.removeEventListener( 'resize', resize );
+	} );
 
 	// Content size
 	const contentSize = useInitialContentSize( contentRef );
+
+	const refresh = () => {
+		let anchor = computeAnchorRect(
+			anchorRefFallback,
+			anchorRect,
+			getAnchorRect,
+			anchorRef,
+			shouldAnchorIncludePadding
+		);
+
+		if ( ! anchor ) {
+			return;
+		}
+
+		anchor = addBuffer(
+			anchor,
+			anchorVerticalBuffer,
+			anchorHorizontalBuffer
+		);
+
+		const {
+			popoverTop,
+			popoverLeft,
+			xAxis,
+			yAxis,
+			contentHeight,
+			contentWidth,
+		} = computePopoverPosition( anchor, contentSize, position );
+
+		// Compute the animation position
+		const yAxisMapping = {
+			top: 'bottom',
+			bottom: 'top',
+		};
+		const xAxisMapping = {
+			left: 'right',
+			right: 'left',
+		};
+		const animateYAxis = yAxisMapping[ yAxis ] || 'middle';
+		const animateXAxis = xAxisMapping[ xAxis ] || 'center';
+
+		containerRef.current.setAttribute( 'data-x-axis', xAxis );
+		containerRef.current.setAttribute( 'data-y-axis', yAxis );
+		containerRef.current.style.top = popoverTop + 'px';
+		containerRef.current.style.left = popoverLeft + 'px';
+
+		if ( contentHeight ) {
+			contentRef.current.style.maxHeight = contentHeight + 'px';
+		}
+
+		if ( contentWidth ) {
+			contentRef.current.style.maxWidth = contentWidth + 'px';
+		}
+
+		if ( xAxis === 'center' && yAxis === 'middle' ) {
+			contentRef.current.classList.add( 'is-without-arrow' );
+		} else {
+			contentRef.current.classList.remove( 'is-without-arrow' );
+		}
+
+		setAnimateOrigin( animateXAxis + ' ' + animateYAxis );
+	};
+
 	useEffect( () => {
-		if ( contentSize ) {
-			setIsReadyToAnimate( true );
+		if ( isMobileViewport && expandOnMobile ) {
+			return;
 		}
-	}, [ contentSize ] );
-
-	// Compute the position
-	const popoverPosition = usePopoverPosition(
-		anchor,
-		contentSize,
-		position,
-		expandOnMobile,
-		contentRef
-	);
+
+		if ( ! contentSize ) {
+			return;
+		}
+
+		refresh();
+
+		/*
+		 * There are sometimes we need to reposition or resize the popover that
+		 * are not handled by the resize/scroll window events (i.e. CSS changes
+		 * in the layout that changes the position of the anchor).
+		 *
+		 * For these situations, we refresh the popover every 0.5s
+		 */
+		const intervalHandle = window.setInterval( refresh, 500 );
+		window.addEventListener( 'resize', refresh );
+		window.addEventListener( 'scroll', refresh, true );
+
+		return () => {
+			window.clearInterval( intervalHandle );
+			window.removeEventListener( 'resize', refresh );
+			window.removeEventListener( 'scroll', refresh, true );
+		};
+	} );
```

Comment:
Do we want to trigger a refresh on each render?

Reply:
I'll add a list of dependencies, but it's pretty much everything :) Still may be worth it to skip the functions.